### PR TITLE
Remove implicit org Dangerfiles

### PR DESCRIPTION
**Pull request by orta**
It's a bandage over a bug for https://github.com/danger/danger.systems/pull/81 - deeper discussion of this is in #415 

---

* **BREAKING** Removes the implicit Dangerfile support. - orta

  The implicit support was a feature that would check for an `[org]/danger`
  repo, and automatically parse that Dangerfile. Think it was a bit too magic,
  and the only person who I know is using it, said they think it should have 
  been this way from the start. I'm cool with this.

  To handle the replacement, we've added a new object to the DSL.
  This is the `danger` API. It has two responsibilities at the moment,
  downloading a Dangerfile:

  ```ruby
  danger.import_dangerfile = "artsy/danger"
  ```

  and importing local plugins:

  ```ruby
  danger.import_plugin [path_or_url]
  ```

  Which the astute may have remembered used to be the purview of `plugin.import`.
  Which is now removed, in favour of the `danger.import_plugin`. Think there's more
  space for improvement inside the context of `danger` than `plugin`.

  I also removed `plugin.download` - couldn't see a use for it inside a Dangerfile. Happy
  to change that, if it's being used it.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ## Master
 
+* **BREAKING** Removes the implicit Dangerfile work. This is a WIP. The feature will be back in some form. - orta
```
Comment (KrauseFx) [inline CHANGELOG.md]: From this it's not 100% clear on what feature got removed

Comment (orta) [inline CHANGELOG.md]: yeah, it looks like this PR will have to provide all the docs and the replacement - so this will get a real changelog entry

**Comment by orta** [comment]
I'm seeing:

```
NoMethodError: undefined method `danger_repo?' for #<Danger::RequestSources::GitHub:0x007fef42bd3b00>
  /Users/orta/.gem/ruby/2.1.3/bundler/gems/danger-c3c17f273728/lib/danger/commands/local.rb:120:in `check_and_run_org_dangerfile'
  /Users/orta/.gem/ruby/2.1.3/bundler/gems/danger-c3c17f273728/lib/danger/commands/local.rb:110:in `run'
  /Users/orta/.gem/ruby/2.1.3/gems/claide-1.0.0/lib/claide/command.rb:334:in `run'
  /Users/orta/.gem/ruby/2.1.3/bundler/gems/danger-c3c17f273728/bin/danger:5:in `<top (required)>'
  /Users/orta/.gem/ruby/2.1.3/bin/danger:23:in `load'
  /Users/orta/.gem/ruby/2.1.3/bin/danger:23:in `<top (required)>'
```

**Comment by orta** [comment]
Danger is also dying on circle CI master - https://circleci.com/gh/danger/danger.systems/22?utm_campaign=build-failed&utm_medium=email&utm_source=notification

**Pull request merged**